src/sentry/static/sentry/app/components/button.jsx

```diff
@@ -0,0 +1,59 @@
+import jQuery from 'jquery';
+import React from 'react';
+import classNames from 'classnames';
+
+import LoadingIndicator from '../components/LoadingIndicator';
+
+const Button = React.createClass({
+  propTypes: {
+    progress: React.PropTypes.bool,
+    disabled: React.PropTypes.bool,
+    icon: React.PropTypes.string
+  },
+
+  getDefaultProps() {
+    return {
+      progressBar: false,
+      disabled: false,
+    };
+  },
+  getInitialState() {
+    return {
+      isInitiated: false,
+      isComplete: false,
+    };
+  },
+
+  click() {
+    this.setState({isInitiated: true});
+    setTimeout(this.setState.bind(this, {isComplete: true}), 500);
```

Comment:
So this is technically unsafe, since this entire component may be destroyed before the timer ticks, making this error when `this` doesn't exist.

Correct me if I'm wrong @benvinegar 

Also, not a fan of just blindly waiting 500ms without leveraging _any_ real feedback.